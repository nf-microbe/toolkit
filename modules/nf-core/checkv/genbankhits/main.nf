process CHECKV_GENBANKHITS {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/pyhmmer_pandas:9f16bf144bcd5b1a':
        'community.wave.seqera.io/library/pyhmmer_pandas:842847931f77e2ab' }"

    input:
    tuple val(meta) , path(aai)
    path ncbi
    path db

    output:
    tuple val(meta), path ("${prefix}_checkv_genbank_hits.tsv") , emit: genbank_hits
    path "versions.yml"                                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    checkv_genbank_hits.py \\
        --checkv_aai ${aai} \\
        --checkv_db ${db} \\
        --ncbi_tsv ${ncbi} \\
        --output ${prefix}_checkv_genbank_hits.tsv \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_checkv_genbank_hits.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """
}
