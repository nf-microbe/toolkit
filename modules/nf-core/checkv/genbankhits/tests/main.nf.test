nextflow_process {

    name "Test Process CHECKV_GENBANKHITS"
    script "../main.nf"
    process "CHECKV_GENBANKHITS"

    tag "modules"
    tag "modules_nfcore"
    tag "checkv"
    tag "checkv/endtoend"
    tag "checkv/downloaddatabase"
    tag "checkv/genbankhits"

    setup {
        run("CHECKV_DOWNLOADDATABASE") {
            script "../../downloaddatabase/main.nf"
        }
    }

    test("bacteroides_fragilis - checkv - aai.tsv") {
        when {
            process {
                """
                input[0] = [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/genome/tsv/genome_checkv_aai.tsv", checkIfExists: true)
                ]
                input[1] = "${projectDir}/assets/db/ncbi_info.tsv"
                input[2] = CHECKV_DOWNLOADDATABASE.out.checkv_db
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("checkv - empty aai.tsv") {
        when {
            process {
                """
                input[0] = [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/genome/tsv/empty_aai.tsv", checkIfExists: true)
                ]
                input[1] = "${projectDir}/assets/db/ncbi_info.tsv"
                input[2] = CHECKV_DOWNLOADDATABASE.out.checkv_db
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/genome/tsv/genome_checkv_aai.tsv", checkIfExists: true)
                ]
                input[1] = "${projectDir}/assets/db/ncbi_info.tsv"
                input[2] = CHECKV_DOWNLOADDATABASE.out.checkv_db
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
