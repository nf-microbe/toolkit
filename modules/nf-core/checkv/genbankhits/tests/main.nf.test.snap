{
    "checkv - empty aai.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_checkv_genbank_hits.tsv:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c490ce4ec802d615af992dae067ffddf"
                ],
                "genbank_hits": [
                    [
                        {
                            "id": "test"
                        },
                        "test_checkv_genbank_hits.tsv:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c490ce4ec802d615af992dae067ffddf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-25T12:04:46.306386231"
    },
    "bacteroides_fragilis - checkv - aai.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_checkv_genbank_hits.tsv:md5,269c2a0fb27c12bfe0de255af33b6cb1"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c490ce4ec802d615af992dae067ffddf"
                ],
                "genbank_hits": [
                    [
                        {
                            "id": "test"
                        },
                        "test_checkv_genbank_hits.tsv:md5,269c2a0fb27c12bfe0de255af33b6cb1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c490ce4ec802d615af992dae067ffddf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T16:52:57.180285654"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_checkv_genbank_hits.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c490ce4ec802d615af992dae067ffddf"
                ],
                "genbank_hits": [
                    [
                        {
                            "id": "test"
                        },
                        "test_checkv_genbank_hits.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c490ce4ec802d615af992dae067ffddf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T16:53:17.899766681"
    }
}